app/lib/nhs/immunisations_api.rb

```diff
@@ -187,12 +194,77 @@ def should_be_in_immunisations_api?(
     )
       vaccination_record.kept? && vaccination_record.recorded_in_service? &&
         vaccination_record.administered? &&
-        vaccination_record.programme.type.in?(PROGRAMME_TYPES) &&
+        vaccination_record.programme.type.in?(WRITE_PROGRAMME_TYPES) &&
         (ignore_nhs_number || vaccination_record.patient.nhs_number.present?) &&
         vaccination_record.notify_parents &&
         vaccination_record.patient.not_invalidated?
     end
 
+    def search_immunisations(patient, programmes:, date_from: nil, date_to: nil)
+      unless Flipper.enabled?(:immunisations_fhir_api_integration) &&
+               Flipper.enabled?(:immunisations_fhir_api_integration_search)
+        Rails.logger.info(
+          "Not searching for vaccination records in the immunisations API as one of the" \
+            " feature flags is disabled: Patient #{patient.id}"
+        )
+        return
+      end
+
+      if programmes.empty?
+        raise "Cannot search for vaccination records in the immunisations API; no programmes provided."
+      elsif !programmes.all? do |programme|
+            programme.type.in?(READ_PROGRAMME_TYPES)
+          end
+        raise "Cannot search for vaccination records in the immunisations API; one or more programmes is not supported."
+      end
+
+      Rails.logger.info(
+        "Searching for vaccination records in immunisations API for patient: #{patient.id}"
+      )
+
+      params = {}
+      params[
+        "patient.identifier"
+      ] = "https://fhir.nhs.uk/Id/nhs-number|#{patient.nhs_number}"
+      params["-immunization.target"] = programmes.map(
+        &:snomed_target_disease_name
+      ).join(",")
+      # Format: YYYY-MM-DD
+      params["-date.from"] = date_from.strftime("%F") if date_from
+      params["-date.to"] = date_to.strftime("%F") if date_to
```

Comment:
```suggestion
      params = {
        "patient.identifier" => "https://fhir.nhs.uk/Id/nhs-number|#{patient.nhs_number}",
        "-immunization.target" => programmes.map(&:snomed_target_disease_name).join(","),
        "-date.from" => date_from&.strftime("%F"),
        "-date.to" => date_to&.strftime("%F")
      }.compact
```

I think this would be a more typical Ruby way of populating the hash.